sound/soc/sof/topology.c

```diff
@@ -1041,6 +1041,15 @@ static int sof_control_load_volume(struct snd_soc_component *scomp,
 		goto out;
 	}
 
+	/*
+	 * If control has more than 2 channels we need to override the info. This is because even if
+	 * ASoC layer has defined topology's max channel count to SND_SOC_TPLG_MAX_CHAN = 8, the
+	 * pre-defined dapm control types (and related functions) creating the actual control
+	 * restrict the channels only to mono or stereo.
```

Comment:
so the question will be: why not fix the ASoC topology core, instead of doing an SOF-specific fix?

Intel is pretty much the only user of the topology, so it'd be odd not to fix what we depend on?

Reply:
@plbossart its a bit of a chicken and egg kind of situation. topolopgy expects one of the three kinds of controls (mixer, enum or bytes) and this is determined based on the info op ID. So, bespoke info IDs are not allowed. But adding a bespoke info ID is also hard because we dont know what type of control to load in this case.

What I think might be a better thing to do would be to introduce a new type of info called SND_SOC_TPLG_CTL_VOLSW_MULTI_CHANNEL or something like that and use that in tplg. This would require changes in alsalib as well. @juimonen what do you think?

Reply:
is there a reason why SND_SOC_TPLG_CTL_VOLSW would be limited to 2ch?